WPMedia/fusion-news-theme-blocks
# Tmedia 288 card list react optimizations

## Description
Reduce dom nodes and maintain style. Use images properly sized

## Jira Ticket
- [TMEDIA-288](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-288&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria
Optimize for reducing dom nodes without layout regressions

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-288-card-list-react-optimizations`
2. Use engine theme sdk branch https://github.com/WPMedia/engine-theme-sdk/pull/376
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/card-list-block`
3. Go to http://localhost/pf/homepage/?_website=the-gazette and scroll down to see card list. Compare with corecomponents
![Screen Shot 2021-06-25 at 12 20 42](https://user-images.githubusercontent.com/5950956/123462303-c077c580-d5af-11eb-937c-ac561573f3bf.png)
4. Use lighthouse to measure how many dom nodes

## Effect Of Changes
### Before
<img width="784" alt="Screen Shot 2021-06-25 at 09 39 14" src="https://user-images.githubusercontent.com/5950956/123462359-d8e7e000-d5af-11eb-9c9b-2dc132d1ff92.png">
https://corecomponents.arcpublishing.com/pf/card-list-many-elements/?_website=the-sun

### After
<img width="818" alt="Screen Shot 2021-06-25 at 12 04 09" src="https://user-images.githubusercontent.com/5950956/123462373-de452a80-d5af-11eb-8b2f-5ac06843b30c.png">
- create page with just card list and "the-sun" site. and add a query for 50 type:story to replicate the live site 
- see comments for perf improvements https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-288

## Dependencies or Side Effects
- Removed hr elements because it does not reflect "a thematic break between paragraph-level elements" and we want to optimize for reducing dom nodes. I recreated the hr visually element with css; some browsers may experience this differently
- Also removed an unused div that had no styles (only a class name, at least styles from us) associated with it

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
